diff --git a/custom_components/homeconnect_ws/__init__.py b/custom_components/homeconnect_ws/__init__.py
index a5b224b..d5c4b54 100644
--- a/custom_components/homeconnect_ws/__init__.py
+++ b/custom_components/homeconnect_ws/__init__.py
@@ -171,13 +171,13 @@ async def async_setup_entry(
     appliance = coordinator.appliance
     device_info = DeviceInfo(
         connections={(CONNECTION_NETWORK_MAC, format_mac(appliance.info["mac"]))},
-        hw_version=appliance.info["hwVersion"],
+        hw_version="",
         identifiers={(DOMAIN, appliance.info["deviceID"])},
         name=f"{appliance.info['brand'].capitalize()} {appliance.info['type']}",
         manufacturer=appliance.info["brand"].capitalize(),
         model=f"{appliance.info['type']}",
         model_id=appliance.info["vib"],
-        sw_version=appliance.info["swVersion"],
+        sw_version="",
     )
 
     available_entities = get_available_entities(appliance)
diff --git a/custom_components/homeconnect_ws/config_flow.py b/custom_components/homeconnect_ws/config_flow.py
index 6ff1d21..d003891 100644
--- a/custom_components/homeconnect_ws/config_flow.py
+++ b/custom_components/homeconnect_ws/config_flow.py
@@ -85,6 +85,9 @@ def process_zip_file(config_path: Path) -> dict[str, dict[str, dict | DeviceDesc
             feature_file = profile_file.open(feature_file_name).read()
 
             appliance_description = parse_device_description(description_file, feature_file)
+            appliance_info |= appliance_description['info']
+            appliance_info['deviceID'] = appliance_info['serialNumber']
+            appliance_description['info'] = appliance_info
             appliances[appliance_info["haId"]] = {
                 "info": appliance_info,
                 "description": appliance_description,
